sigma/plugins/fun/xkcd.py

```diff
@@ -2,24 +2,36 @@
 import aiohttp
 import discord
 
-
 async def xkcd(cmd, message, args):
 
-    comic_no = str(random.randint(1, 2171))
+    random_comic = True
     if args:
         if not is_int(args[0]):
             await message.channel.send(None, embed=discord.Embed(title=':exclamation: Invalid number', color=0x993333))
             return
-        
+
         comic_no = args[0]
-        
-    joke_url = 'http://xkcd.com/' + comic_no + '/info.0.json'
-    async with aiohttp.ClientSession() as session:
-        async with session.get(joke_url) as data:
-            try: joke_json = await data.json()
-            except Exception:
-                await message.channel.send(None, embed=discord.Embed(title=':exclamation: Requested xkcd doesn\'t exist', color=0x993333))
-                return
+        random_comic = False
+
+    while True:
+        if not random_comic:
+            joke_url = 'http://xkcd.com/' + comic_no + '/info.0.json'
+        else:
+            joke_url = 'http://xkcd.com/info.0.json'
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get(joke_url) as data:
+                try:
+                    joke_json = await data.json()
```

Comment:
code styling; I put single liners on same line as the `try`

Reply:
Fixed with 6fe84e9

Reply:
Check commit link; that seems to for a different thing

Reply:
Oops.

Fixed with 92135b4